diff --git a/source/slang-glslang/CMakeLists.txt b/source/slang-glslang/CMakeLists.txt
index ed1935d..14ad229 100644
--- a/source/slang-glslang/CMakeLists.txt	
+++ b/source/slang-glslang/CMakeLists.txt
@@ -16,10 +16,11 @@ if(SLANG_ENABLE_SLANG_GLSLANG)
         # allocator is thread-local), and shared inputs such as the resource-limit
         # table are read-only.
         SKIP_SANITIZERS
-        LINK_WITH_PRIVATE glslang SPIRV SPIRV-Tools-opt SPIRV-Tools-link
+        LINK_WITH_PRIVATE glslang SPIRV SPIRV-Tools-opt SPIRV-Tools SPIRV-Tools-link
         INCLUDE_DIRECTORIES_PRIVATE
             ${slang_SOURCE_DIR}/include
             ${CMAKE_CURRENT_BINARY_DIR}/../slang/slang-version-header
+            ${CMAKE_INSTALL_FULL_INCLUDEDIR}/glslang
         INSTALL
         EXPORT_SET_NAME SlangTargets
     )
