diff --git a/CMakeLists.txt b/CMakeLists.txt
index 302eeed..7db8d9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,8 +67,4 @@ add_executable(hyprshutdown ${SRCFILES})
 
 target_link_libraries(hyprshutdown PkgConfig::deps glaze::glaze)
 
-install(
-  FILES contrib/hyprshutdown.desktop
-  DESTINATION "share/applications")
-
-install(TARGETS hyprshutdown)
\ No newline at end of file
+install(TARGETS hyprshutdown)
