diff --git a/src/state/AppState.cpp b/src/state/AppState.cpp
index 3dbe202..5829cbc 100644
--- a/src/state/AppState.cpp
+++ b/src/state/AppState.cpp
@@ -49,7 +49,7 @@ void CApp::quit() {
             return;
         }
         g_logger->log(LOG_TRACE, "CApp::quit: using close for {}", m_class);
-        auto ret = HyprlandIPC::getFromSocket(std::format("/dispatch closewindow address:{}", m_address));
+        auto ret = HyprlandIPC::getFromSocket(std::format("/dispatch hl.dsp.window.close({{ window = 'address:{}' }})", m_address));
         if (!ret)
             g_logger->log(LOG_ERR, "Failed closing window {}: ipc err", m_class);
         else if (*ret != "ok")
diff --git a/src/ui/UI.cpp b/src/ui/UI.cpp
index 73b7576..2884a1b 100644
--- a/src/ui/UI.cpp
+++ b/src/ui/UI.cpp
@@ -216,7 +216,7 @@ void CUI::exit(bool closeHl) {
 
         if (closeHl && !m_noExit && !State::state()->m_dryRun) {
             //NOLINTNEXTLINE
-            HyprlandIPC::getFromSocket("/dispatch exit");
+            HyprlandIPC::getFromSocket("/dispatch hl.dsp.exit()");
             if (m_postExitCmd) {
                 CProcess proc("/bin/sh", {"-c", m_postExitCmd.value()});
                 proc.runAsync();
